### ISSUE TITLE
Rails 5.2.2: ActiveSupport::Duration#inspect returns invalid result

### ISSUE THREAD

**Issue by yivo** [rails/rails#35443]
### Steps to reproduce
```ruby
require "active_support/all"

(5.minutes + 100.seconds).inspect
```

### Expected behavior
`6 minutes and 40 seconds`

### Actual behavior
`5 minutes and 100 seconds`

### System configuration
**Rails version**: 5.2.2

**Ruby version**: 2.6.1

**Comment by st0012** [comment]
This happens on `6.0.0.beta1` as well

**Comment by josepegea** [comment]
This issue is addressed by #35197 , too, which is been open for some time already
